### ISSUE TITLE
meta(script): tracking issue for script improvements

### ISSUE THREAD

**Issue by mds1** [foundry-rs/foundry#4444]
### Component

Forge

### Describe the feature you would like

WIP

**Comment by odyslam** [comment]
Based on our experience in making multiple deployments in December 2022, a big takeaway was that the transaction lifecycle management needed rework. 

Due to unreliable RPC providers **and** network issues, the foundry script should be more robust in handling these issues and issuing the proper transactions in the appropriate order. 

The TL;DR is that the forge script should be more robust and recover from more failures without the user resorting to multiple `--resume` executions.

**Comment by sambacha** [comment]
> Based on our experience in making multiple deployments in December 2022, a big takeaway was that the transaction lifecycle management needed rework.
> 
> Due to unreliable RPC providers **and** network issues, the foundry script should be more robust in handling these issues and issuing the proper transactions in the appropriate order.
> 
> Relevant issues are being discussed afaik in the `ethers-rs` telegram, so I understand that the required work is part `foundry script`, part `ethers-rs`.
> 
> The TL;DR is that the forge script should be more robust and recover from more failures without the user resorting to multiple `--resume` executions.

Why not just let users submit bundles instead of individual transactions? That removes a lot of issues.

No need to recover if atomic.

**Comment by mds1** [comment]
I agree that would be a nice feature, but it only helps on mainnet when the total bundle uses <30M gas, so it doesn’t solve the problem for all users and all chains

**Comment by sambacha** [comment]
> I agree that would be a nice feature, but it only helps on mainnet when the total bundle uses <30M gas, so it doesn’t solve the problem for all users and all chains

Ethereum mainnet is the relevant chain. Supporting submission via bundles can be adopted downstream by other chains.

Transaction submission bundling isn't about gas usage per se, it's about atomic deployment. 

The case about other users and other chains is irrelevant, as it does not worsen their situation.
